x-pack/plugins/apm/server/routes/service_map/get_service_map_service_node_info.ts

```diff
@@ -215,7 +215,11 @@ async function getTransactionStats({
     params
   );
 
-  const totalRequests = response.hits.total.value;
+  const throughputValue = response.aggregations?.timeseries.buckets
+    .map((bucket) => bucket.doc_count)
+    .reduce((prev, current) => {
+      return prev + current;
+    }, 0);
```

Comment:
I don't mind this, but you can also use lodash's sumBy method here. it's a little more explicit and less lines of code.

Reply:
You could also use a `value_count` aggregation: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-valuecount-aggregation.html. It is intended to count the number of values and will actually iterate over the histogram. However it is more efficient to do it as you're doing now because you're re-using data from an aggregation that we already need 👍 . Just keep in mind that this only works on counts, not on averages (as you would need a weighted average), and the buckets have to cover the entire data set. E.g., a terms bucket with a size of 10 might mean you miss data.